core/src/main/java/hudson/tasks/LogRotator.java

```diff
@@ -217,7 +216,10 @@ public void perform(Job<?,?> job) throws IOException, InterruptedException {
             //Collate all encountered exceptions into a single exception and throw that
             String msg = String.format(
                     "Failed to rotate logs for [%s]",
-                    Joiner.on(", ").join(exceptionMap.keySet())
+                    exceptionMap.keySet().stream().
+                        filter(run -> run!=null).
+                        map(Run::toString).
+                        collect(Collectors.joining(", "))
```

Comment:
This code is fine. But perhaps a simpler alternative might be to use `hudson.Util`:

```java
Util.join(exceptionMap.keySet(), ", ")
```

I don't feel strongly one way or another.